src/jdk.compiler/share/classes/com/sun/tools/javac/code/Type.java

```diff
@@ -877,7 +877,9 @@ public boolean isExtendsBound() {
                 kind == UNBOUND;
         }
         public boolean isUnbound() {
-            return kind == UNBOUND;
+            // is it `?` or `? extends Object`?
+            return kind == UNBOUND ||
+                    (kind == EXTENDS && type.hasTag(CLASS) && ((ClassType)type).supertype_field == Type.noType);
```

Comment:
For the test on whether `type` is j.l.Object, would something like this work better?
```suggestion
                    (kind == EXTENDS && type.tsym.flatName() == type.tsym.name.table.names.java_lang_Object);
```

Just an idea. (Sadly, we can't use `Symtab` here, that would be ideal.)